extensions/3rdparty/LyricWiki/LyricFind/LyricFindTrackingService.class.php

```diff
@@ -251,6 +251,29 @@ public static function isPageBlockedViaApi($amgId="", $gracenoteId="", $pageTitl
 		wfProfileOut(__METHOD__);
 		return $isBlocked;
 	}
+
+	public static function isWebCrawler( $userAgent ) {
+		// A list of some common words used only for bots and crawlers.
+		$crawlers = [
+			'bot',
+			'slurp',
+			'crawler',
+			'spider',
+			'curl',
+			'facebook',
+			'fetch',
+			'ruby',
+			'python'
```

Comment:
The top UA's we're being hit by are things like `perl/lyrics::fetcher::lyricwiki 0.10` `mozilla/5.0 (windows; u; windows nt 5.1; en-us; rv:1.8.1.6) gecko/20070725 firefox/2.0.0.6` `mozilla/5.0 (x11; linux x86_64) applewebkit/535.19 (khtml, like gecko) chrome/18.0.1025.142 safari/535.19`

Trying to blacklist these by UA seems fragile

Reply:
👍